.changeset/late-books-compete.md

```diff
@@ -0,0 +1,9 @@
+---
+"@pnpm/npm-resolver": patch
+"@pnpm/core": patch
+"pnpm": patch
+---
+
+Deduplicate direct dependencies.
+
+Let's say there are two projects in the workspace that dependend on `foo`. One project has `foo@1.0.0` in the dependencies while another one has `foo@^1.0.0` in the dependencies. In this case, `foo@1.0.0` should be installed to both projects as satisfies the version specs of both projects.
```

Comment:
Wanted to briefly comment on yarn's `--strategy` argument. I think the pnpm behavior before mostly matches yarn's existing (and only) `highest` strategy. It sounds like the new strategy could be described by `most-matches`.

I was a bit on the fence whether `pnpm dedupe` should prefer a most matches strategy since it could cause some resolutions to downgrade to a version before a security fix. But this is probably fine since the vulnerable version probably should be in the `pnpm-lock.yaml` file in the first place.